### PR TITLE
Changed uin8 to uint8 in response to issue #9266

### LINKED ISSUE #9266: uin8 should be uint8 in io/parsers.py
while using read_fwf I had a bug that had this line from parsers.py

https://github.com/pydata/pandas/blob/072e40b0b5beab36c2adda9e0bcbb26755e1928f/pandas/io/parsers.py#L989

in which there is a np.uin8 which I'm pretty sure should be a np.uint8

apologise if this has already been submitted - I did a search on uin8 and only got the offending line in the code.

I just fixed on my arch source code to keep working on my problem (I'm an engineer not a software guy!). I will try and find time for a patch and test (I failed on my last one)

CHeers

Rick

### PR DESCRIPTION
**Pull request by CoolRanch29**
closes #9266

### CODE REVIEW

**Comment by shoyer** [comment]
Thanks, this needs:

1. a test case which breaks before this change (you can use the example from #9266; add it to `pandas/io/tests/test_parsers.py`)
2. an addition the "bug fixes" section of `docs/sources/whatsnew/0.16.0.txt` (refer to the original issue)

**Comment by CoolRanch29** [comment]
This is my first PR attempt and I don't understand this error message.  Did I do something wrong in the test function I added?

File: pandas/io/tests/test_parsers.py [hunk 1]
```diff
@@ -2984,6 +2984,28 @@ def test_variable_width_unicode(self):
         tm.assert_frame_equal(expected, read_fwf(BytesIO(test.encode('utf8')),
                                                  header=None, encoding='utf8'))
 
+    def test_convert_to_nd_arrays(self):
+        #GH 9266
+        with open('test.txt','w') as f:
+            f.write(
+            """1421302964.213420    PRI=3 PGN=0xef00      DST=0x17 SRC=0x28    04 154 00 00 00 00 00 127
+               1421302964.226776    PRI=6 PGN=0xf002               SRC=0x47    243 00 00 255 247 00 00 71"""
```
Comment (shoyer) [inline pandas/io/tests/test_parsers.py]: I think the issue is that this line starts with extra whitespace, throwing off `colspecs` below.

**Comment by jreback** [comment]
@CoolRanch29 I rebase and push your branch up here. pls review and update this PR when you can
https://github.com/jreback/pandas/tree/CoolRanch29-master

I am not sure what you are trying to compare against here (e.g. what should the expected frame look like?)

**Comment by CoolRanch29** [comment]
Okay, I added an expected frame to test again.  The problem before is that pd.read_fwf wasn't working at all - an exception was thrown when trying to read the test case.  I'm still concerned that in the resulting frame, the column 'dst' is of type 'object' when using read_fwf and of type 'float64' when created by DataFrame().

File: doc/source/whatsnew/v0.16.0.txt [hunk 2]
```diff
@@ -209,4 +209,9 @@ Bug Fixes
 - Fixed issue in the ``xlsxwriter`` engine where it added a default 'General' format to cells if no other format wass applied. This prevented other row or column formatting being applied. (:issue:`9167`)
 - Fixes issue with ``index_col=False`` when ``usecols`` is also specified in ``read_csv``. (:issue:`9082`)
 - Bug where ``wide_to_long`` would modify the input stubnames list (:issue:`9204`)
+<<<<<<< HEAD
+
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.0.txt]: needs fixing here (eliminate the merge lines)

File: pandas/io/tests/test_parsers.py [hunk 3]
```diff
@@ -2984,6 +2984,31 @@ def test_variable_width_unicode(self):
...
+        result = pd.read_fwf(StringIO(test),
+        header=False,
+        colspecs=[(0,17),(25,26),(33,37),(49,51),(58,62),(63,1000)],
+        names=['time','pri','pgn','dst','src','data'],
+        converters={'pgn':lambda x: int(x,16),
+        'src':lambda x: int(x,16),
+        'dst':lambda x: int(x,16),
+        'data':lambda x: len(x.split(' '))},
+        index_col='time')
+        expected = pd.DataFrame([[ 3, 61184, 23, 40, 8] ,
```
Comment (jreback) [inline pandas/io/tests/test_parsers.py]: this doesn't look property formatted to me

**Comment by jreback** [comment]
whats new is now available for 0.16.1 https://github.com/pydata/pandas/commit/5ebf521eac528f5a4b772e4677f70b689b8b1826

**Comment by jreback** [comment]
can you rebase this?

**Comment by CoolRanch29** [comment]
I'm pretty sure I just rebased.  I'm sorry if this is causing a headache for you, I'm sure I'm doing some things stupidly.  I've gotten a lot of benefit out of using pandas and so I'm trying to contribute but this is my first time contributing so I'm having to learn a lot about both the pandas code and github.

File: pandas/io/tests/test_parsers.py [hunk 4]
```diff
@@ -2984,6 +2984,28 @@ def test_variable_width_unicode(self):
         tm.assert_frame_equal(expected, read_fwf(BytesIO(test.encode('utf8')),
                                                  header=None, encoding='utf8'))
 
+    def test_convert_to_nd_arrays(self):
+        #GH 9266
+        with open('test.txt','w') as f:
```
Comment (jreback) [inline pandas/io/tests/test_parsers.py]: use
```
with tm.ensure_clean('test.txt') as path:
     with open(path,'w') as f:
          ....
```

File: pandas/io/tests/test_parsers.py [hunk 5]
```diff
@@ -2984,6 +2984,28 @@ def test_variable_width_unicode(self):
...
+            f.write(
+                """1421302964.213420    PRI=3 PGN=0xef00      DST=0x17 SRC=0x28    04 154 00 00 00 00 00 127 \n"""
+            +   """1421302964.226776    PRI=6 PGN=0xf002               SRC=0x47    243 00 00 255 247 00 00 71"""
+                )
+        try:
```
Comment (jreback) [inline pandas/io/tests/test_parsers.py]: do 

```
result = pd.read_fwf(......)
self.assertEqual(result['dst'].dtype,np.uint8)
```

(I for whichever columns should be that dtype)

**Comment by jreback** [comment]
couple of style things. pls add a release note in v0.16.1 as well.

**Comment by jreback** [comment]
We'd like to merge this. Pls address the comments from above. (and needs a rebase). see [here](http://pandas-docs.github.io/pandas-docs-travis/contributing.html) for help on this.

**Comment by jreback** [comment]
@CoolRanch29 can you update

**Comment by jreback** [comment]
@CoolRanch29 want to update?

**Comment by CoolRanch29** [comment]
Okay, I believe I just rebased.

File: doc/source/whatsnew/v0.16.1.txt [hunk 6]
```diff
@@ -80,3 +80,5 @@ Bug Fixes
 
 - Bug in ``Series.quantile`` on empty Series of type ``Datetime`` or ``Timedelta`` (:issue:`9675`)
 - Bug in ``where`` causing incorrect results when upcasting was required (:issue:`9731`)
+
+- Bug in ``ParserBase.convert_to_nd_arrays`` when called by ``DataFrame.read_fwf`` (:issue:`9266``)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.1.txt]: needs to be moved to 0.17.0. Say something like ``fixed bug in csv parsing when using a converting that specified uint8``

**Comment by jreback** [comment]
See here for rebasing help. Pls squash to a single commit as well. http://pandas.pydata.org/pandas-docs/stable/contributing.html

**Comment by jreback** [comment]
can you rebase/squash

**Pull request closed without merging**